CONTRIBUTING.md

```diff
@@ -47,6 +47,19 @@ Please make sure you have signed the [Contributor License Agreement](http://www.
   ./bin/kibana --dev
   ```
 
+#### `config/kibana.dev.yml`
+
+The `config/kibana.yml` file stores user configuration directives but, since this file is checked into source control, developer preferences can't be saved without the risk of accidentally committing the modified version. To make customizing configuration easier during development the kibana CLI will look for a `config/kibana.dev.yml` file if run with the `--dev` flag.
```

Comment:
I think it's worth mentioning that settings you put here override the default config as well